app.rb

```diff
@@ -0,0 +1,126 @@
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'user_inputs'
+
+class App
+  attr_reader :books, :person, :rentals, :classroom
+
+  def initialize
+    @books = []
+    @person = []
+    @rentals = []
+  end
+
+  # Methods to list books and people
+  def list_all_books()
+    puts 'All Books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people()
+    puts 'All People:'
+    @person.each do |person|
+      puts "#{person.class}, Name: #{person.name}, age: #{person.age}"
+
+      if person.instance_of?(Teacher)
+        puts "Specialization: [#{person.specialization}]"
+      else
+        puts "Permission: #{person.parent_permission}"
+      end
+    end
+  end
+
+  def create_person()
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    is_student = UserInput.new.self_get
+    puts 'Enter person name:'
+    name = gets.chomp
+    puts 'Enter person age:'
+    age = UserInput.new.self_get
+
+    case is_student
+    when 1
+      puts 'Does student have parent permission [Y/N]: '
+      permission = UserInput.new.user_permission
+      student = Student.new(1, age, permission, name)
+      @person << student
+
+    when 2
+      print 'What is the teachers specialization: '
+      specialization = gets.chomp
+      teacher = Teacher.new(age, specialization, name)
+      @person << teacher
+    end
+
+    puts 'Person created successfully.'
+  end
+
+  # Creating book method
+
+  def create_book
+    puts 'Enter book title:'
+    title = gets.chomp
+    puts 'Enter book author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
```

Comment:
- [ ] This piece of code does not adhere to the single responsibility rule, and also contravenes the KISS principles as all requests for inputs could have been placed elsewhere like the `main.rb` file and not here
#### NB: Please reference this [lesson on best practices SOLID principles](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)